src/chains-config/cache/QueryFeeDetailsCache.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * When checking the cache there are 3 possible choices to return.
+ */
+enum QueryFee {
+	/**
+	 * queryFeeDetails is available
+	 */
+	available = 'available',
+	/**
+	 * queryFeeDetails is not available
+	 */
+	notAvailable = 'notAvailable',
+	/**
+	 * We dont know is queryFeeDetails is available
+	 */
+	unknown = 'unknown',
+}
+
+export class QueryFeeDetailsCache {
```

Comment:
this runtime API is called `TransactionPaymentCallApi` but no sure if we really care about `query_call_info` as well

Reply:
Yea we don't care for query_call_info here. Only query_fee_details

Reply:
We only want to cache the versions of QueryFeeDetails

Reply:
I was thinking of having some nice abstraction of the `TransactionPaymentApi` but if we only care about `query_call_info` then this is sufficient.

Reply:
fixed